plugins/modules/ec2_asg.py

```diff
@@ -220,6 +220,13 @@
       - When I(propagate_at_launch) is true the tags will be propagated to the Instances created.
     type: list
     elements: dict
+  purge_tags:
+    description:
+      - If C(true), existing tags will be purged from the resource to match exactly what is defined by I(tags) parameter.
+      - If the I(tags) parameter is not set then tags will not be modified.
+    default: true
```

Comment:
Forgive me for commenting on an already merged PR, but I just found out about this from a complete breakage of all ansible AWS services that use tags.

Why was `default=True` chosen, for a new field that no one was using previously, deemed a good idea?

Now, everything that touches any asset that uses tags MUST now add a `purge_tags: False` just to leave things as they are??

Otherwise all tags are deleted???

Existing use case to scale in/out an ASG, and touch nothing else:
```
- community.aws.ec2_asg:
    name: "{{ item.auto_scaling_group_name }}"
    min_size: "{{ count }}"
    max_size: "{{ count }}"
    desired_capacity: "{{ count }}"
    region: eu-central-1
```

Now, the above and strips all tags as well. 

Given that most ASG tags are used to propagate context information to the ec2 instances, this now destroys all context information for the ASG and the instances it creates have no information about the runtime context.

How is that the "principle of least surprising behavior"?

This now requires adding the purge_tags: False everywhere where tags could be impacted.
```
- community.aws.ec2_asg:
    name: "{{ item.auto_scaling_group_name }}"
    min_size: "{{ count }}"
    max_size: "{{ count }}"
    desired_capacity: "{{ count }}"
    purge_tags: False       <----- This is now needed everywhere tags are used to say "leave alone" ?
    region: eu-central-1
```

This is the ONLY case where one has to specify a non-default option just to say: "leave alone".

Note also that
```
    purge_tags: False 
```
is NOT backwards compatible, because that field didn't exist in previous versions, so you need 2 different cases depending on which side of this version you fall on.

Given that the `purge_tags` things seems to have propagated to many AWS assets, all in slightly different releases, one needs to account for this on a case by case basis depending on when the `purge_tags` was introduced into each module.

Now, if the default had been `purge_tags=False`, none of this would be an issue. 
There would have been no changes required as the new behavior would match the old.

Am I the only one complaining about this? I can't believe this didn't hit more people.
Does no use use ansible to alter "only what is specified" and expect the rest to remain as-is?

I can't decide now whether to walk away from ansible as an AWS "minor edit" tool or, try deal with this "maybe we'll change something that destroys something else later, that you can't know about until it happens.